app/waf/src/controllers/declaration.controller.ts

```diff
@@ -333,4 +342,56 @@ export class DeclarationController extends BaseController {
       .declarations(applicationId)
       .delete({and: [{id: declarationId}, {tenantId: await this.tenantId}]});
   }
+
+  @post(
+    prefix +
+      '/applications/{applicationId}/declarations/{declarationId}/deploy',
+    {
+      responses: {
+        '204': Schema.emptyResponse('Successfully deploy Application resource'),
+        '404': Schema.notFound('Can not find Application resource'),
+        '422': Schema.unprocessableEntity(
+          'Fail to deploy Application resource',
+        ),
+      },
+    },
+  )
+  async deployById(
+    @param(Schema.pathParameter('applicationId', 'Application resource ID'))
+    applicationId: string,
+    @param(Schema.pathParameter('declarationId', 'Declaration resource ID'))
+    declarationId: string,
+  ): Promise<string> {
+    let declarations = await this.applicationRepository
+      .declarations(applicationId)
+      .find({
+        where: {
+          and: [
+            {
+              id: declarationId,
+            },
+            {
+              tenantId: await this.tenantId,
+            },
+          ],
+        },
+      });
+
+    if (declarations.length === 0) {
+      throw new HttpErrors.NotFound('Cannot find Declaration');
+    }
+
+    let application = await this.applicationRepository.findById(applicationId);
+
+    if (!application.adcId) {
+      throw new HttpErrors.UnprocessableEntity(
+        'No target ADC to perform deploy action',
+      );
+    }
+
+    let adc = await this.adcRepository.findById(application.adcId);
```

Comment:
do we need to select by tenantId here?

Reply:
BTY,  can we replace find with findById at #L367?